src/pstl_offload.cpp

```diff
@@ -0,0 +1,182 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <new>
+#include <cassert>
+#include <cstdint>
+#include <sycl/sycl.hpp>
+
+#include <pstl_offload/internal/usm_memory_replacement_common.h>
+
+#define _PSTL_OFFLOAD_VERSION_MAJOR 1
+#define _PSTL_OFFLOAD_VERSION_MINOR 0
+#define _PSTL_OFFLOAD_VERSION_PATCH 0
```

Comment:
What kind of a version is this intended to be? Only the version of the binary, or really the version of the whole implementation? Depending on that, we may need to change either the name or the placement of the macros.

Reply:
Yes.

Reply:
Done